lua/telescope/config.lua

```diff
@@ -282,6 +282,30 @@ local telescope_defaults = {
     { "rg", "--color=never", "--no-heading", "--with-filename", "--line-number", "--column", "--smart-case" },
   },
 
+  smart_mask = { false },
+
+  smart_mask_args = {
```

Comment:
Both should not be config values. config is meant for options across more than one picker.

Reply:
Mind elaborating on this? This is excellent for me to understand. If it's not a config value, how does the user pass the table to the code?

Reply:
So we have this system.
```lua
require("telescope").setup {
  defaults = { 
    -- your defaults for telescope
  },
  pickers = {
    live_grep = {
      -- your defaults for live_grep
      smart_mask = {
        ["lua"] = "-tlua",
        ["cs"]  = "-tcs",
        ["cpp"] = "-tcpp",
        ["h"]   = "-tcpp"
      }
    }
  }
}
```
So if you now call `require('telescope.builtin').livegrep()` it will applies these `live_grep` defaults. If you then do `require('telescope.builtin').livegrep({ smart_mask = {} })` it would prefer the opts again.

So for your code. You only need to check for `opts.smart_mask`

Reply:
The one thing i dont know how to handle yet would be the documentation. The documentation needs to happen in `builtin/init.lua` again but our docgen system can only understand single line annotations.

Reply:
Awesome, this was the missing piece for me. Much better now. Thanks!

Reply:
PR updated. Anything I can do to help with docs?